docs/_docs/02_features/debugging.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: collection-browser-doc
+title: Debugging
+category: features
+categories_url: features
+excerpt: Learn how to debug issues with terragrunt and terraform.
+tags: ["DRY", "Use cases", "CLI"]
+order: 220
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+## Debugging
+
+Terragrunt and Terraform usually play well together in helping you
+write DRY, re-usable infrastructure. But how do we figure out what
+went wrong in the rare case that they _don't_ play well?
+
+Terragrunt provides a debug mode you can access through the `--terragrunt-debug`
+command flag. For example you could use it like this to debug an `apply`
+that's producing unexpected output:
+
+    $ terragrunt --terragrunt-debug apply
+
+Running this command will do two things for you:
+  - Output a file named `terragrunt-debug.tfvars` to your terraform working
+    directory (the same one containing your `terragrunt.hcl` or `main.tf`
+    terraform code)
+  - Print instructions on how to invoke terraform against the generated file to
+    reproduce exactly the same terraform output as you saw when invoking
+    `terragrunt`. This will help you to understand whether the problem is coming
+    from
+
+The flag's goal is to help you determine which of these three major areas is the
+root cause of your problem:
+  1. Misconfiguration of your infrastructure code.
+  2. An error in `terragrunt`.
+  3. An error in `terraform`.
+  
+
+Let's run through a few use-cases.  
+
+### Use-case: I use locals or dependencies in terragrunt.hcl, and the terraform output isn't what I expected
+ 
+Consider this file structure for a fictional production environment where we
+have configured an application to deploy as many tasks as there are minimum
+number of machines in some cluster.
+
+```
+    └── live
+        └── prod
+            └── app
+            |   ├── vars.tf
+            |   ├── main.tf
+            |   ├── outputs.tf
+            |   └── terragrunt.hcl
+            └── ecs-cluster
+                └── outputs.tf
+
+```
+
+The files contain this text (`app/main.tf` and `ecs-cluster/outputs.tf` elided
```

Comment:
NIT: Good word! New to me. Consider using "omitted" as it is more conventional for such an example.

Reply:
Done